src/electron-main.ts

```diff
@@ -193,8 +192,8 @@ async function setupGlobals(): Promise<void> {
 
     // The tray icon
     // It's important to call `path.join` so we don't end up with the packaged asar in the final path.
-    const iconFile = `element.${process.platform === 'win32' ? 'ico' : 'png'}`;
-    iconPath = path.join(resPath, "img", iconFile);
+    const iconPath = global.vectorConfig.tray_icon_path ||
+      path.join(resPath, "img", `element.${process.platform === 'win32' ? 'ico' : 'png'}`);
```

Comment:
As per Electron docs, ICO is recommended on Windows, this configuration model doesn't support this.

> On Windows it is recommended to use ICO icons to get best visual effects

https://www.electronjs.org/docs/latest/api/browser-window
https://www.electronjs.org/docs/latest/api/tray

Reply:
Are you referring to chaning the current behavior to also choose ico forsomething like `win32`..?

Otherwise I think this can be solved by the builder/build system, just make sure to configure and include ico for windows builds and png otherwise.

Reply:
EDIT: Looks like it's `win32` across the boards for windows so you must be referring to the latter.

Reply:
The two options I see would be to either have a `tray_icon_prefix` where it'd be sans the file-ending, and have it inferred and appended, or to split it up into two separate vars. I'm not really a fan of either idea here.

Reply:
an object for formats isn't awful, I think. 

```json
{
  "tray_icons": {
    "png": "/path/to/element.png",
    "ico": "/path/to/element.ico"
  }
}
```

then if you don't specify the one it needs, it uses the default.

Reply:
Fair enough, that's reasonable!

Reply:
Code is updated. Realized now that consequence of this approach is that users who want to customize the icon and have portable config across OSs will now need to maintain their icons in the formats, whereas with the previous approach they could just use png.

But I guess that every scenario can't be the happy path and it's a rare enough edge case with a minor enough inconvenience that it's OK.